src/axom/sidre/core/Group.hpp

```diff
@@ -889,6 +894,79 @@ class Group
 
   //@}
 
+  //@{
+  //!  @name Accesors for group and view collections.
+  //!
+  //! These methods can be used to iterate on the collection of groups and views
+  //! Example:
+  //!      for (auto& group : grp->groups())
+  //!      {
+  //!          /// code here using group
+  //!      }
+  //!      for (auto& view : grp->views())
+  //!      {
+  //!          /// code here using view
+  //!      }
+
+  /*!
+   * \brief Returns a reference to the collection of views
+   */
+  ViewCollection& views() { return *m_view_coll; }
+
+  /*!
+   * \brief Returns a const reference to the collection of views
+   */
+  const ViewCollection& views() const { return *m_view_coll; }
+
+  /*!
+   * \brief Returns a reference to the collection of groups
+   */
+  GroupCollection& groups() { return *m_group_coll; }
+
+  /*!
+   * \brief Returns a const reference to the collection of groups
+   */
+  const GroupCollection& groups() const { return *m_group_coll; }
+
+  //@}
+private:
+  /*!
+   * \brief Casts the views ItemCollection to a (named) MapCollection
+   *
+   * \warning This is only valid when the group is using a map rather than a list
+   * for its collection of views
+   * \sa isUsingMap, isUsingList
+   */
+  MapCollection<View>* getNamedViews();
```

Comment:
Expose pointer to non-const?

Reply:
These functions are private, so I think they should be ok.

They're only there to cast from the base type `ItemCollection` to the derived type `MapCollection` (with an assert that it is ok to do so).  The cast is now necessary because I removed functionality related to named items from the base class.

Reply:
Sorry. I missed the private part.